### ISSUE TITLE
In training process, using the test set to generate the negative samples! 

### ISSUE THREAD

**Issue by pp413** [KXY-PUBLIC/NASE#1]
Thanks for sharing!
There is still a question in your releasing source codes, eg.,  in the line 36-37 of dataloader.py file, you have used the test set and valid set for training your model! After  correcting,  the final results will be bad! For example, Hist@10 < 40%.
